src/main/java/org/takes/rs/RsPrettyXML.java

```diff
@@ -114,10 +118,53 @@ private Response make() throws IOException {
                 OutputKeys.OMIT_XML_DECLARATION, "yes"
             );
             transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+            if (isHtml5(source.getInputSource().getByteStream())) {
+                // @checkstyle MultipleStringLiteralsCheck (1 line)
+                transformer.setOutputProperty(OutputKeys.METHOD, "html");
+                transformer.setOutputProperty(OutputKeys.VERSION, "5.0");
+            }
             transformer.transform(source, new StreamResult(result));
         } catch (final TransformerException ex) {
             throw new IOException(ex);
         }
         return result.toByteArray();
     }
+
+    /**
+     * Checks if the input is an html5 document.
+     * @param body The body to be checked.
+     * @return True if the input is an html5 document.
+     * @checkstyle MethodNameCheck (4 lines)
+     */
+    @SuppressWarnings({ "PMD.AvoidCatchingGenericException",
+        "PMD.OnlyOneReturn" })
+    private static boolean isHtml5(final InputStream body) {
+        final Document document;
+        try {
+            body.mark(Integer.MAX_VALUE);
+            document = parseBody(body);
+            body.reset();
+            // @checkstyle IllegalCatchCheck (1 line)
+        } catch (final Exception ex) {
+            return false;
+        }
+        final DocumentType doctype = document.getDoctype();
+        return doctype != null
+            && "html".equalsIgnoreCase(doctype.getName())
+            && doctype.getSystemId() == null
+            && doctype.getPublicId() == null;
+    }
+
+    /**
+     * Parses the input stream and returns the Document built.
+     * @param body The body to be parsed.
+     * @return The document built.
+     * @throws Exception if something goes wrong.
+     */
+    private static Document parseBody(final InputStream body) throws Exception {
```

Comment:
@carlosmiranda ok done. Even though I prefer having a separate method to improve readability